facade_charm/fetch_interfaces_from_charmhub.py

```diff
@@ -0,0 +1,61 @@
+"""Scrape charmhub to grab all interfaces for all registered charms.
+"""
+import asyncio
+import json
+import logging
+from pathlib import Path
+
+import aiohttp
+import requests
+import tenacity
+import yaml
+
+logger = logging.getLogger("update-endpoints")
+
+FACADE_CHARM_ROOT = Path(__file__).parent
+CH_INTERFACES_PATH = FACADE_CHARM_ROOT / 'charmhub_interfaces.yaml'
+
+
+def _get_all_registered_charms():
+    print('fetching store...')
+    resp = requests.get("https://charmhub.io/packages.json")
+    return resp.json()['packages']
```

Comment:
Consider adding `resp.raise_for_status()`;

or possibly configuring the client to automatically raise for bab status values.

Reply:
OT: this gives me exactly 360 charms.

I wonder if that's all there is, or is it some kind of server limit?